src/display/ui_state.rs

```diff
@@ -1,35 +1,77 @@
-use ::std::collections::{BTreeMap, HashMap};
+use ::std::collections::{BTreeMap, HashMap, VecDeque};
 use ::std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 
 use crate::network::{Connection, LocalSocket, Utilization};
 
+static BANDWIDTH_DECAY_FACTOR: f32 = 0.5;
+static RECALL_LENGTH: usize = 5;
+
 pub trait Bandwidth {
     fn get_total_bytes_downloaded(&self) -> u128;
     fn get_total_bytes_uploaded(&self) -> u128;
+
+    fn get_avg_bytes_downloaded(&self) -> u128;
+    fn get_avg_bytes_uploaded(&self) -> u128;
 }
 
 #[derive(Default)]
 pub struct NetworkData {
     pub total_bytes_downloaded: u128,
     pub total_bytes_uploaded: u128,
+    pub prev_total_bytes_downloaded: u128,
```

Comment:
Now these fields are not used anymore, perhaps we should remove them?

Reply:
Ah yeah - for sure. Didn't want to cause too much of a mess in case you were in the middle of something.

I saw you've removed this already though, so thanks :)